vdr/src/contracts/cl/types/schema.rs

```diff
@@ -17,6 +17,46 @@ pub struct Schema {
     pub attr_names: Vec<String>,
 }
 
+impl Schema {
+    pub fn id(&self) -> SchemaId {
+        SchemaId::build(&self.issuer_id, &self.name, &self.version)
+    }
+
+    pub fn matches_id(&self, expected_id: &SchemaId) -> VdrResult<()> {
+        let actual_id = self.id();
+
+        if expected_id.to_string() != actual_id.to_string() {
+            return Err(VdrError::InvalidSchema(format!(
+                "Id built from schema: {} != provided id: {}",
```

Comment:
I think we can delete this method (and for cred def) as it's called from a single place. and move check there

Reply:
Deleted